src/actions/ReplaceWithText.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../Types";
+import displayPendingEditDecorations from "../editDisplayUtils";
+import { runForEachEditor } from "../targetUtils";
+import { flatten, zip } from "lodash";
+import { maybeAddDelimiter } from "../getTextWithPossibleDelimiter";
+import { performEditsAndUpdateSelections } from "../updateSelections";
+
+export default class implements Action {
```

Comment:
really cool how short this impl is with all the new abstractions

Reply:
Yes those was really needed. Turned out great!